quickwit/quickwit-opentelemetry/src/otlp/mod.rs

```diff
@@ -119,46 +120,52 @@ pub(crate) fn extract_attributes(attributes: Vec<OtlpKeyValue>) -> HashMap<Strin
         if let Some(value) = attribute
             .value
             .and_then(|any_value| any_value.value)
-            .and_then(to_json_value)
+            .and_then(oltp_value_to_json_value)
         {
             attrs.insert(attribute.key, value);
         }
     }
     attrs
 }
 
-fn to_json_value(value: OtlpValue) -> Option<JsonValue> {
+fn oltp_value_to_json_value(value: OtlpValue) -> Option<JsonValue> {
     match value {
         OtlpValue::ArrayValue(OtlpArrayValue { values }) => Some(
             values
                 .into_iter()
-                .flat_map(to_json_value_from_primitive_any_value)
+                .flat_map(|value| value.value)
+                .flat_map(oltp_value_to_json_value)
```

Comment:
can we have filter_map here? It makes it clearer that this never generate "new" elements (this implementation made me think `value.value` was a Vec and i couldn't figure out why it would be that)